cmd/git-security/api/repository_test.go

```diff
@@ -132,3 +139,240 @@ func TestArchiveRepoNormal(t *testing.T) {
 	require.Nil(t, err)
 	assert.Equal(t, 200, resp.StatusCode)
 }
+
+func TestSetSession(t *testing.T) {
+	app := fiber.New()
+
+	store := session.New(session.Config{
+		Expiration:   time.Hour,
+		KeyLookup:    "cookie:session_id",
+		KeyGenerator: utils.UUID,
+	})
+
+	// Handler function to set session
+	app.Get("/set-session", func(c *fiber.Ctx) error {
+		sess, err := store.Get(c)
+		if err != nil {
+			return err
+		}
+		sess.Set("username", "testuser")
+		if err := sess.Save(); err != nil {
+			return err
+		}
+		return c.SendStatus(fiber.StatusOK)
+	})
+
+	// Handler function to get session
+	app.Get("/get-session", func(c *fiber.Ctx) error {
+		sess, err := store.Get(c)
+		if err != nil {
+			return err
+		}
+		username := sess.Get("username")
+		return c.JSON(fiber.Map{"username": username})
+	})
+
+	// Create a new request to set the session
+	req := httptest.NewRequest("GET", "/set-session", nil)
+	resp, err := app.Test(req, -1)
+	require.Nil(t, err)
+	assert.Equal(t, fiber.StatusOK, resp.StatusCode)
+
+	// Extract session ID from the response cookies
+	cookies := resp.Header["Set-Cookie"]
+	require.NotEmpty(t, cookies)
+
+	// Create a new request to get the session
+	req = httptest.NewRequest("GET", "/get-session", nil)
+	for _, cookie := range cookies {
+		req.Header.Add("Cookie", cookie)
+	}
+	resp, err = app.Test(req, -1)
+	require.Nil(t, err)
+	assert.Equal(t, fiber.StatusOK, resp.StatusCode)
+
+	// Read and parse the response body
+	body, err := io.ReadAll(resp.Body)
+	require.Nil(t, err)
+
+	var result map[string]string
+	err = json.Unmarshal(body, &result)
+	require.Nil(t, err)
+
+	// Verify the session data
+	assert.Equal(t, "testuser", result["username"])
+}
+
+func TestGetRepositoriesCSVs(t *testing.T) {
```

Comment:
Test is too long without focus (your focus should be on that repository api) , and then you didnt reproduce the problem of duplicated columns ... you just had the erorr in the for loop ... we need to work on this together .